docs/developer/decision-records/2022-06-28-entities-timestamp/README.md

```diff
@@ -0,0 +1,449 @@
+        contract_policy_id VARCHAR NOT NULL,
+        created_timestamp BIGINT,
+        selector_expression JSON NOT NULL,
+        PRIMARY KEY(contract_definition_id)
+        );
+```
+
+Interface `contractDefinitionStatements`
+
+```java
+public interface ContractDefinitionStatements {
+    ...
+
+    default String getCreatedTimestamp() {
+        return "created_timestamp";
+    }
+    ...
+}
+```
+
+Class `BaseSqlDialectStatements`
+
+```java
+public class BaseSqlDialectStatements implements ContractDefinitionStatements {
+    ...
+
+    @Override
+    public String getInsertTemplate() {
+        return format("INSERT INTO %s (%s, %s, %s, %s, %s) VALUES (?, ?, ?, ?, ?%s)",
+                getContractDefinitionTable(),
+                getIdColumn(),
+                getAccessPolicyIdColumn(),
+                getContractPolicyIdColumn(),
+                getCreatedTimestamp(),
+                getSelectorExpressionColumn(),
+                getFormatAsJsonOperator());
+    }    
+    ...
+}
+```
+
+### Contract Agreement
+
+The changes related to the entity `ContractAgreement` are the following:
+
+Class `ContractAgreement`
+
+```java
+public class ContractAgreement {
+    ...
+    private long createdTimestamp;
+    ...
+
+    private ContractAgreement(@NotNull String id,
+                              @NotNull String providerAgentId,
+                              @NotNull String consumerAgentId,
+                              long contractSigningDate,
+                              long contractStartDate,
+                              long contractEndDate,
+                              @NotNull Policy policy,
+                              @NotNull String assetId,
+                              long createdTimestamp) {
+        ...
+        this.createdTimestamp = createdTimestamp;
+    }    
+    ...
+
+    /**
+     * Creation timestamp of the {@link ContractAgreement}.
+     *
+     * @return contract id
+     */
+    @NotNull
+    public long getCreatedTimestamp() {
+        return createdTimestamp;
+    }
+    ...
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static class Builder {
+
+        private String id;
+        private long createdTimestamp;
+        ...
+
+        public Builder createdTimestamp(long createdTimestamp) {
+            this.createdTimestamp = createdTimestamp;
+            return this;
+        }
+        ...
+
+        public ContractAgreement build() {
+            return new ContractAgreement(id, providerAgentId, consumerAgentId, contractSigningDate, contractStartDate, contractEndDate, policy, assetId, createdTimestamp);
+        }
+        ...
+    }
+}
+```
+
+file `schema.sql`
+
+```roomsql
+...
+CREATE TABLE IF NOT EXISTS edc_contract_agreement
+(
+    agr_id            VARCHAR NOT NULL
+        CONSTRAINT contract_agreement_pk
+            PRIMARY KEY,
+    provider_agent_id VARCHAR,
+    consumer_agent_id VARCHAR,
+    created_timestamp BIGINT,
+    signing_date      BIGINT,
```

Comment:
Isn't the created date the same as the signing date?

Reply:
@florianrusch-zf no, they aren't the same, they have two different meanings: 
one represents the time the entity was created, the other is a domain value that represent the time the agreement was signed. Currently they have the same value but the meaning is quite different.
Moreover, `createdTimestamp` is in milliseconds, `signingDate` in epoch seconds